DSharpPlus/Net/DiscordApiClient.cs

```diff
@@ -425,6 +425,68 @@ internal async Task<DiscordInvite> GetGuildVanityUrlAsync(ulong guild_id)
 
             return invite;
         }
+
+        internal async Task<DiscordWidget> GetGuildWidgetAsync(ulong guild_id)
+        {
+            var route = $"{Endpoints.GUILDS}/:guild_id{Endpoints.WIDGET_JSON}";
+            var bucket = this.Rest.GetBucket(RestRequestMethod.GET, route, new { guild_id }, out var path);
+
+            var url = Utilities.GetApiUriFor(path);
+            var res = await this.DoRequestAsync(this.Discord, bucket, url, RestRequestMethod.GET).ConfigureAwait(false);
+
+            var ret = JsonConvert.DeserializeObject<DiscordWidget>(res.Response);
+            ret.Discord = this.Discord;
+            ret.Guild = Discord.Guilds[guild_id];
+
+            var json = JObject.Parse(res.Response);
+            var rawChannels = (JArray)json["channels"];
+            ret.Channels = rawChannels.Select(r =>
+            {
+                DiscordChannel c = ret.Guild.GetChannel(Convert.ToUInt64(r["id"]));
```

Comment:
There may be times where the guild cache is null, so there should be a check here that passes the partial channel object provided from the response if `ret.Guild` is null.

Also for casting the channel id, you can do `(ulong)r["id"];` rather than using `Convert` to keep things consistent with the lib.